include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -319,22 +623,29 @@ class PreconditionerBoomerAMG : public PreconditionerBase<Number>
 /**
  * Implementation of AMG preconditioner unifying PreconditionerML and PreconditionerBoomerAMG.
  */
-template<typename Operator, typename Number>
+template<int dim, typename Operator, typename Number>
 class PreconditionerAMG : public PreconditionerBase<Number>
 {
 private:
   typedef typename PreconditionerBase<Number>::VectorType VectorType;
 
 public:
-  PreconditionerAMG(Operator const & pde_operator, bool const initialize, AMGData const & data)
+  PreconditionerAMG(Operator const &             pde_operator,
+                    bool const                   initialize,
+                    AMGData const &              data,
+                    dealii::Mapping<dim> const & mapping)
   {
     (void)pde_operator;
     (void)initialize;
+    (void)mapping;
     this->data = data;
 
     if(data.amg_type == AMGType::BoomerAMG)
     {
 #ifdef DEAL_II_WITH_PETSC
+      AssertThrow(data.amg_operator_type == AMGOperatorType::Default,
+                  dealii::ExcMessage("AMGType::BoomerAMG incorporates "
+                                     "only AMGOperatorType::Default."));
```

Comment:
This check implies that one has to change `amg_operator_type` in the input file when switching from `Trilinos` to `PETSc` for AMG. Principally, I prefer if one does not have to do this. Alternatives would be (i) to simply ignore this parameter for `PETSc` and document in the code that this parameter is only used for `Trilinos`, or (ii) to indicate by the parameter name that this parameter is for `Trilinos` only, e.g. `ConstantModeTypeTrilinos`.

Reply:
I will remove the assert and adjust the name.